upgrades/steps_28.go

```diff
@@ -72,3 +88,103 @@ func prepopulateRebootHandledFlagsForDeployedUnits(ctx Context) error {
 	}
 	return nil
 }
+
+func moveUniterStateToController(ctx Context) error {
+	// Lookup the names of all unit agents installed on this machine.
+	agentConf := ctx.AgentConfig()
+	_, unitNames, _, err := service.FindAgents(agentConf.DataDir())
+	if err != nil {
+		return errors.Annotate(err, "looking up unit agents")
+	}
+	if len(unitNames) == 0 {
+		// No units, nothing to do.
+		return nil
+	}
+
+	var fileNames []string
+	if ctx.AgentConfig().Value(agent.ProviderType) == k8sprovider.CAASProviderType {
+		if fileNames, err = caasOperatorLocalState(ctx, unitNames); err != nil {
+			return err
+		}
+	} else {
```

Comment:
nested if can be hoisted to the parent else

Reply:
agreed, however in this case will make the code harder to read.